### PR TITLE
docs: Code Search with Vector Embeddings Using Qdrant

### PR DESCRIPTION
**Pull request by Anush008**
## Description

Adds a notebook demonstrating a [semantic code-search](https://code-search.qdrant.tech/) approach with vector embeddings and Qdrant.

### CODE REVIEW

File: notebooks/en/code_search.ipynb [hunk 1]
```diff
@@ -0,0 +1,1130 @@
...
+        ").points\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "k2vbMhW2QQCm"
+      },
+      "source": [
+        "Now, review the results. The following table lists the module, the file name\n",
```
Comment (Anush008) [inline notebooks/en/code_search.ipynb:580]: TODO: Rerun these queries. Not all vectors where indexed.

File: notebooks/en/code_search.ipynb [hunk 2]
```diff
@@ -0,0 +1,1130 @@
...
+        "id": "4QXn9c57QQCl"
+      },
+      "source": [
+        "Let's begin generating embeddings for our data. We'll use [FastEmbed](https://github.com/qdrant/fastembed) - A CPU-first, lightweight library for generating vector embeddings."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "%pip install https://github.com/qdrant/fastembed/archive/jina-embeddings-v2-base-code.zip"
```
Comment (Anush008) [inline notebooks/en/code_search.ipynb:342]: TODO: Investigate the disparity found with SenteceTransformers

File: notebooks/en/code_search.ipynb [hunk 3]
```diff
@@ -0,0 +1,1130 @@
+{
```
Comment (merveenoyan) [inline notebooks/en/code_search.ipynb:1]: <p>if you want you can install these libraries at the top of the notebook and only do the imports where necessary</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (merveenoyan) [inline notebooks/en/code_search.ipynb:1]: <p>we could end on an enthusiastic tone, call for action build and give useful resources (maybe qdrant docs?)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

**Review by merveenoyan — COMMENTED** [review]
Mostly readability nits, LGTM otherwise :)

**Comment by merveenoyan** [comment]
@Anush008 can you add your notebook to index.md?

**Comment by Anush008** [comment]
Hey @merveenoyan. Thanks for the review. I'll act on it soon.
I am currently re-trying the outputs since I found some discrepancies.

**Comment by merveenoyan** [comment]
I'd like @stevhliu to skim through it before we merge ☺️ otherwise looks good to me

File: notebooks/en/code_search.ipynb [hunk 4]
```diff
@@ -0,0 +1,779 @@
...
```
Comment (stevhliu) [inline notebooks/en/code_search.ipynb:1]: <p>You can put the NOTE in a special Tip block like this:</p><p> </p>

<pre class="ql-syntax" spellcheck="false">&gt; [!TIP]
&gt; In-memory can only be used for quick prototyping and tests.
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (stevhliu) [inline notebooks/en/code_search.ipynb:1]: <p>Same comment for this NOTE here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

**Review by stevhliu — APPROVED** [review]
Nice job!

File: notebooks/en/code_search.ipynb (hunk 4 above)
Comment (Anush008) [reply notebooks/en/code_search.ipynb:1]: Done in [`f82bf10` (#151)](https://github.com/huggingface/cookbook/pull/151/commits/f82bf10b7fd32c4c26139395fbbce70b7eb6eb00).

File: notebooks/en/code_search.ipynb [hunk 5]
```diff
@@ -0,0 +1,781 @@
...
```
Comment (stevhliu) [inline notebooks/en/code_search.ipynb:1]: <p>The doc-builder only supports <code>TIP</code> and <code>WARNING</code> so <code>IMPORTANT</code> won't work</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/151/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (Anush008) [reply notebooks/en/code_search.ipynb:1]: Thanks, Steven. Fixed [`f9fbbb1` (#151)](https://github.com/huggingface/cookbook/pull/151/commits/f9fbbb1dd5aa64181390e0f7541419b8db14d9ae)

**Comment by Anush008** [comment]
Hey @stevhliu. I don't see the notebook in the Cookbook. Is it not instantaneous?

**Comment by Anush008** [comment]
Scratch that. I see it now at https://huggingface.co/learn/cookbook/en/code_search.

**Pull request merged**